### DISCUSSION TITLE
Ist it best-practices to distribute control-plane nodes across multiple locations?

### DISCUSSION THREAD

**Discussion by DENKweit** [mysticaltech/terraform-hcloud-kube-hetzner#1264] [Q&A] [answered]
Basically title. 

I was just wondering, since this is the advertised default setting in kube.tf.

What are the downsides of distributing the control-plane nodes?

**Comment by WebSpider** [comment, marked as answer]
You typically want your control-plane as highly available as possible, since without a control-plane, no changes in pods can occur, including spawning new ones.

The biggest downside is that whenever the cloud provider has maintenance or an outage, the chance of something being down is higher than when you run everything together, however the impact is lower.

**Reply by mysticaltech** [reply]
Exactly! And always in odd numbers, and at least 3.

**Comment by maaft** [comment]
What about latency? I heard that highish latency can also disrupt a cluster.

maybe @mysticaltech  can also give is opinion?
